app/models/semester.rb

```diff
@@ -19,15 +18,18 @@ class Semester < ActiveRecord::Base
 
   validates :year, uniqueness: { scope: [:season] }, presence: true
 
-  def can_be_destroyed?
-    ! (student_applications.exists? || projects.exists?)
+  def self.create_current_semester
```

Comment:
Maybe this name is a bit misleading since the semester it creates is not the current semester in in the site but rather the current semester in real life? What about `self.create_using_current_date` ? Naming is always the hardest part.